graspy/align/seedless_procrustes.py

```diff
@@ -0,0 +1,373 @@
+# Copyright 2020 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ot
+import numpy as np
+from sklearn.utils import check_array
+
+from .base import BaseAlign
+from .sign_flips import SignFlips
+from .orthogonal_procrustes import OrthogonalProcrustes
+
+
+class SeedlessProcrustes(BaseAlign):
+    """
+    Implements an algorithm that matches two datasets using an orthogonal
+    matrix. Unlike OrthogonalProcrustes, this does not use a matching between
+    entries. It can even be used in the settings when the two datasets do not
+    have the same number of vertices.
+
+    In essence, it aims to simultaneously obtain a, not necessarily 1-to-1,
+    correspondance between the vertices of the two data sets, and the
+    orthogonal matrix alignment matrix. It does via a sequence of alternating
+    steps, similarly to a regular expectation-maximization procedures. The step
+    of obtaining an assignment ("E-step") is done by solving an optimal
+    transport problem via Sinkhorn algorithm, whereas obtaining an orthogonal
+    alignment matrix ("M-step") is done via regular orthogonal procurstes.
+
+    Parameters
+    ----------
+        optimal_transport_lambda : float (default=0.1), optional
+            Regularization term of the Sinkhorn optimal transport algorithm.
+
+        optimal_transport_eps : float (default=0.01), optional
+            Tolerance parameter for the each Sinkhorn optimal transport
+            algorithm. I.e. tolerance for each "E-step".
+
+        optimal_transport_num_reps : int (default=1000), optional
+            Number of repetitions in each iteration of the iterative optimal
+            transport problem. I.e. maximum number of repetitions in each
+            "E-step".
+
+        optimal_transport_eps : float (default=0.1), optional
+            Tolerance for the each iteration of the iterative optimal transport
+            problem. I.e. tolerance of the whole "EM" algorithm.
+
+        iterative_num_reps : int (default=100), optional
+            Number of reps in each iteration of the iterative optimal transport
+            problem. I.e. maxumum number of total iterations the whole "EM"
+            algorithm.
+
+        initialization: string, {"2d" (default), "sign_flips", "custom"}, optional
```

Comment:
i motion to call this `init` for consistency with sklearny stuff (e.g. [kmeans](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html))

Reply:
:+1:

Reply:
addressed.